apis/clusters/v1beta1/validation_test.go

```diff
@@ -0,0 +1,1072 @@
+package v1beta1
+
+import (
+	"testing"
+
+	"github.com/instaclustr/operator/pkg/models"
+)
+
+func TestCluster_ValidateCreation(t *testing.T) {
+	type fields struct {
+		Name                  string
+		Version               string
+		PCICompliance         bool
+		PrivateNetworkCluster bool
```

Comment:
PrivateNetworkCluster is named like PrivateNetwork inside the `GenericClusterSpec` for now